### PR TITLE
feat: use pytest for qemu-system tests

### PR DESCRIPTION
**Pull request by clark-ja**
This PR adds the feature to use `pytest` (including its coverage capabilities) to the `qemu-system` tests. I took the scripts for the userland tests as a template and modified them until things worked as I wanted to.

I also added a few, not very meaningful but exemplary, tests for the kernel commands `kbase`, `kchecksec`, `kcmdline`, `kconfig`, `kversion`, `slab list` and `slab info`.

I have not yet taken a look at the necessary changes to the github workflows yet so I open this PR as a Draft  for now.

Screenshot from 2023-04-21 21-41-29

EDIT: now I also implemented the changes for the github workflow.

The failing test `tests/test_loads.py::test_loads_binary_with_core_without_crashing` on the archlinux docker is unrelated to this PR.

### CODE REVIEW

**Comment by clark-ja** [comment]
lol, the PR decreases the coverage by 56.41%? I guess this is only the coverage for the kernel commands now.

Either I messed something up, or CodeCovs promise does not hold :D
> No setup required. Automatic report merging for all CI and languages into a single report.

EDIT: ah, the report was now corrected automatically...

**Comment by clark-ja** [comment]
I just noticed that it might also be useful to only execute tests for certain architectures or kernel types later on, so I passed them through to the test via env vars.

Conditional tests could be executed e.g. like this:

```python
import os
import pytest

ARCH = os.getenv("PWNDBG_ARCH")
KERNEL_TYPE = os.getenv("PWNDBG_KERNEL_TYPE")

@pytest.skipif(ARCH not in ["x86_64"], reason="")
def test_only_on_x86_64():
    raise Exception("This test is only executed on x86_64")
```

File: tests/qemu-tests/pytests_collect.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+class CollectTestFunctionNames:
+    """See https://github.com/pytest-dev/pytest/issues/2039#issuecomment-257753269"""
+
+    def __init__(self):
+        self.collected = []
+
+    def pytest_collection_modifyitems(self, items):
+        for item in items:
+            self.collected.append(item.nodeid)
```
Comment (disconnect3d) [inline tests/qemu-tests/pytests_collect.py:17]: Or: `self.collected.extend(item.nodeid for item in items)`

**Comment by disconnect3d** [comment]
This is awesome work. Thanks!

**Pull request merged**